src/api/java/com/minecolonies/api/crafting/ItemStorage.java

```diff
@@ -98,11 +114,17 @@ public ItemStorage(@NotNull final ItemStack stack)
     {
         this.stack = stack;
         this.shouldIgnoreDamageValue = false;
-        this.shouldIgnoreNBTValue = false;
+        this.shouldIgnoreNBTValue = getDefaultIgnoreNBT(stack);
```

Comment:
We really can't/shouldn't make certain items get ignored in comparisons of all uses of ItemStorage, afterall it is used in far more places than just crafting. I do not think this is a good way to handle these things

Reply:
It's used in crafting and the request system.  That specific line that you highlighted only takes effect for itemstacks that have been specially tagged, which can only ever happen in recipes so cannot affect anything other than crafting, and doesn't affect any other usage of the same item, so it is entirely safe.

The item tag is more broad and will apply to every usage of that item (which is why I initially didn't have it in until Orion suggested it), but the specific items that are in the tag should always be "safe" to ignore NBT on for all purposes; when they do acquire NBT it's never anything important and just causes incompatibilities with recipes and other usage without good reason.  It's certainly more dangerous, if someone adds inappropriate things to the tag, but hopefully nobody does, or they quickly learn the error of their ways.  But the global nature does make it less likely for these known-problematic items to actually cause problems.

Reply:
It is used a lot in storage e.g. racks aswell as multi-purpose item comparisons ranging through a broad variety of use cases, just slamming an ignore into all of these for one use case doesnt make a lot of sense.

Reply:
Also I dont not think an ignore all nbt of specific items is a good solution of your problem either, I'd rather see a one sided hasRequiredNBt

Reply:
Now, one thing we have to be aware off, and we might want to have a comment on, is how someaddons said it, since we use this for matching, this can result in "stack compression" such that nbt data of items is multiplicated/lost.

I do think that all the items in the tag are totally find with that, but we should probably comment this on the tag

Reply:
Yes, stack compression is definitely a thing, but as Raycoms said for these specific items it should cause no harm.  I'm happy to add a dire warning to the tag JSON to that effect; what's the preferred syntax for that in a way that doesn't break the tag?  (I can invent one that works, but I'd prefer to use an established syntax if there is one.)

Reply:
And FWIW the tag is not intended to solve only this one use case.  These particular items have been perennial thorns in the side of all the use cases, including storage.

Reply:
I'm aware that there are issues caused by our currently complete/exact nbt comparisons, but just plugging the compat code for crafting into an all-purpose comparison storage is just a bad place for it